docs/localization/LOCALIZATION.md

```diff
@@ -74,167 +80,180 @@ export function localizationConfigurationFactory() {
+      loader: translateLoaderProvider,
+    })
+    // ...
   ],
-  ...
+  // ...
   providers: [
     {provide: LOCALIZATION_CONFIGURATION_TOKEN, useFactory: customCreateLocalizationConfiguration, deps: [YourServiceNeededByTheFactory]}
   ]
+})
+class AppModule {}
 ```
 
-You may also need to set up your http server to accept **CORS** and update **connect-src** property of **csp.json** in your application.
+## HTTP server
 
-```typescript
+You may also need to set up your http server to accept **CORS** and update the `connect-src` property of `csp.json` in your application.
 
+```json5
 // configs/csp.json
 
 {
+  "connect-src": "http://example.com"
+}
+```
 
-"connect-src": "http://example.com"
+## Custom localization bundles
 
-}
+If you have your own localization, you need to create your translation bundles in JSON format. Each file is named after a locale and holds the corresponding translations.
+For example, if your application is meant to support French, English and Canadian French, you will have 3 separate files called `en.json`, `fr.json` and `fr-CA.json` (the file name should match the official [locale](https://github.com/angular/angular/tree/master/packages/common/locales) name).
 
+Place your translation bundles either at your dedicated endpoint URL or in your application's `src/assets/assets-otter/i18n` folder. The format of the bundles is a simple key/value pair object,
+where the key is the name of the resource and the value is its translation. 
+
+```json5
+// en.json
+{
+  "o3r-simple-header-pres.motto": "Let's shape the future of travel",
+  "o3r-simple-header-pres.airline": "My Airline",
+  "o3r-simple-header-pres.logoAltText": "Brand Logo Text",
+  "o3r-simple-header-pres.language.en": "English",
+  "o3r-simple-header-pres.language.fr": "Français",
+  "o3r-simple-header-pres.language.ar": "اللغة العربية",
+  "o3r-simple-header-pres.locWithArg": "Hello, {{user}}!"
+}
 ```
 
-- If you have your own localization you need to create your translation bundles in json format. Each file is named after a locale and holds corresponding translations. For example if your application is meant to support French, english and Canadian French you will have 3 separate files called respectively **en.json**, **fr.json** and **fr-CA.json** (the file name should match the official [locale](https://github.com/angular/angular/tree/master/packages/common/locales) name).
+### How to properly create keys?
 
-- Place your translation bundles either at your dedicated endpoint URL or inside your application's folder **src/assets/assets-otter/i18n**. The format of the bundles is a simple key/value pair object where key is the resource name and value its translation. The key naming convention is component selector followed by `.` (dot character) and any string of your choice.
+It's always a good idea to respect certain naming conventions. For components, we prefix each key by the component selector to ensure the uniqueness of cross-components keys. 
```

Comment:
```suggestion
It's always a good idea to respect certain naming conventions. For components, we recommend to prefix each key by the component selector to ensure the uniqueness of cross-components keys. 
```
maybe we can create a linter for that :)

Reply:
👍 I will create an issue for this once the PR is merged

Reply:
https://github.com/AmadeusITGroup/otter/issues/1915